js/node_shared/unix.js

```diff
@@ -0,0 +1,369 @@
+const { spawnSync } = require("node:child_process");
+const os = require("os");
+const fs = require("fs");
+const process = require("process");
+
+globalThis.spawnSync = spawnSync;
+globalThis.fs = fs;
+globalThis.os = os;
+globalThis.process = process;
+
+// mapping pid to its own process
+globalThis.pidToProcessTable = new Map();
+// mapping pid of parent to its children
+globalThis.pidToChildrenTable = new Map();
+// There will be collisions if we ever allocate 20000 or more file descriptors
+// "naturally", that is, through `caml_sys_open`.
+globalThis.fdCount = 20000;
+globalThis.fdTable = new Map();
```

Comment:
Later on I mention something about FDs > 1024, and hoping Node is smart. Disregard that comment because we use 20k here, so I checked. It looks like node uses libuv see [here](https://www.geeksforgeeks.org/libuv-in-node-js/) so this should be fine